sdk/core/core/inc/az_http_transport.h

```diff
@@ -85,12 +85,16 @@ AZ_NODISCARD az_result
 az_http_request_get_header(_az_http_request const* request, int32_t index, az_pair* out_header);
 
 /**
- * @brief Get parts of an HTTP request.
+ * @brief Get parts of an HTTP request. NULL in accepted to ignore getting any parts, for example,
+ * call this function like below to get only the http method and ignore getting url and body
+ *   az_http_request_get_parts(request, &method, NULL, NULL)`
+ *
+ * This function is expected to be used by transport layer only.
  *
  * @param request HTTP request to get parts from.
- * @param out_method Pointer to write HTTP method to.
- * @param out_url Pointer to write URL to.
- * @param out_body Pointer to write HTTP request body to.
+ * @param out_method Pointer to write HTTP method to. Use NULL to ignore getting this value.
+ * @param out_url Pointer to write URL to. Use NULL to ignore getting this value.
+ * @param out_body Pointer to write HTTP request body to. Use NULL to ignore getting this value.
  *
  * @retval AZ_OK Success.
```

Comment:
We should try to be consistent with our docs, especially those returning az_result.
1) Consider using `@return`
2) Consider formatting it as a list, like this:
https://github.com/Azure/azure-sdk-for-c/blob/da98d45fb4a1d44b940cec3e69a9358d566879c7/sdk/core/core/inc/az_span.h#L301-L304

Reply:
BTW, I think we should use `@retval` everywhere we use `@return` as in example above.

Reply:
I am curious to know, why you prefer one over the other? Does it render differently?

Either way, let's try to be consistent.

Reply:
`@retval`'s purpose is exactly to specify each concrete return value: i.e. for `az_result convert_roman_numeral_to_decimal(az_span roman, az_span* out_decimal)`, it can be "`AZ_OK` success, `AZ_ERROR_ARG` input is malformed, `AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY` not enough size to fit the result".

`@return`'s goal is do describe the return value in general, i.e.  for `int strlen(char const* s)` it can be `string length, -1 if s is NULL`.

For the cases when we describe each return value, we happen to format `return` ourselves to describe the return the values, and we may end up matching `@retval`'s output pretty close, but this is what `@retval` is for - it will use Doxygen's method to format them. But anyways, it is readable, but I think ideally it is better to use the specific tool for the job.

Reply:
Interesting. I will take a closer look. Thanks @antkmsft

Reply:
Regarding `@retval` vs `@return`, retval does look pretty nice:
![image](https://user-images.githubusercontent.com/6527137/81455693-be04a300-9144-11ea-9b27-df4ff98e6a7a.png)
```C
/**
 * @brief Appends the beginning of a JSON object (i.e. `{`).
 *
 * @param[in] json_builder A pointer to an #az_json_builder instance containing the buffer to append
 * the start of object to.
 *
 * @return An #az_result value indicating the result of the operation:
 *         - #AZ_OK if object start was appended successfully
 *         - #AZ_ERROR_INSUFFICIENT_SPAN_SIZE if the buffer is too small
 *         - #AZ_ERROR_JSON_NESTING_OVERFLOW if the depth of the JSON exceeds the maximum allowed
 *           depth of 64
 */
AZ_NODISCARD az_result az_json_builder_append_begin_object(az_json_builder* json_builder);

/**
 * @brief Appends the beginning of a JSON array (i.e. `[`).
 *
 * @param[in] json_builder A pointer to an #az_json_builder instance containing the buffer to append
 * the start of array to.
 *
 * @retval #AZ_OK if array start was appended successfully
 * @retval #AZ_ERROR_INSUFFICIENT_SPAN_SIZE if the buffer is too small
 * @retval #AZ_ERROR_JSON_NESTING_OVERFLOW if the depth of the JSON exceeds the maximum allowed
 * depth of 64
 */
AZ_NODISCARD az_result az_json_builder_append_begin_array(az_json_builder* json_builder);
```

Maybe we combine return and retval, and have both?
![image](https://user-images.githubusercontent.com/6527137/81455765-fa380380-9144-11ea-9c6c-2320defa4799.png)